server/alignment_api/pipelines/load_alignments.py

```diff
@@ -0,0 +1,287 @@
+"""
+Usage: python manage.py shell < alignment_api/pipelines/load_alignments.py
+"""
+import os
+import csv
+import json
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+from tqdm import tqdm
+
+from alignment_api.models import SourceToken, TargetToken, Resource, Alignment, Link
+
+DATA_PATH = settings.BASE_DIR / "data/data"
+SOURCES_PATH = DATA_PATH / "sources"
+TARGETS_PATH = DATA_PATH / "targets"
+alignment_targets = ["YLT", "CUVMP"]
+CATALOG = []
+
+
+def load_catalog():
+    CATALOG_PATH = DATA_PATH / "catalog.tsv"
+    with open(CATALOG_PATH) as file:
+        tsv_file = csv.reader(file, delimiter="\t")
+        next(tsv_file)  # skip header
+        for line in tqdm(tsv_file, desc=f"import catalog"):
+            identifier = line[0]
+            format = line[1]
+            alignment_license = line[2]
+            scope = line[3]
+            source_license = line[4]
+            target_license = line[5]
+
+            id_pieces = identifier.split("+")
+            pair = id_pieces[2].split("-")
+
+            new_entry = {}
+            new_entry["lang"] = id_pieces[0]
+            new_entry["source"] = pair[0]
+            new_entry["target"] = pair[1]
+            new_entry["type"] = pair[2]
+            new_entry["alignment_license"] = alignment_license
+            new_entry["source_license"] = source_license
+            new_entry["target_license"] = target_license
+            CATALOG.append(new_entry)
+
+
+def get_alignment_pairs_for_target(target_name):
+    source_oriented_pairs = os.listdir(SOURCES_PATH)
+    target_oriented_pairs = os.listdir(TARGETS_PATH)
+
+
+def determine_source_names_for_target(target_name):
+    source_target_pairs = os.listdir(TARGETS_PATH)
+    source_names = []
+    for source_target_pair in source_target_pairs:
+        cleaned_pair = source_target_pair.split(".")[0].split("-")
+        if cleaned_pair[1] == target_name:
+            source_names.append(cleaned_pair[0])
+    return source_names
+
+
+def get_or_create_resource(resource_name, lang):
+    exists = Resource.objects.filter(name=resource_name).exists()
+    if exists:
+        print(f"\t get {resource_name}")
+        return Resource.objects.get(name=resource_name)
+
+    # TODO: support language ingestion
+    print(f"\t create {resource_name}")
+    new_resource = Resource(name=resource_name, lang=lang)
+    new_resource.save()
+    return new_resource
+
+
```

Comment:
See https://github.com/Clear-Bible/Alignments/blob/main/src/gcsource.py for a dataclass-y way to do this (except for the translation to Django objects).